### PR TITLE
Refactor Operation into AbstractOperation and Swagger2Operation

### PR DESCRIPTION
**Pull request by dtkav**
Relates to #420 

In an attempt to break up the OpenAPI3 PR, this includes all of the refactor that went into that branch, but leaves out other openapi3 support.

Changes proposed in this pull request:
 - Establish interface for what an AbstractOperation does
 - Implements interface with Swagger2Operation
 - Defaults to Swagger2URIParser (*not* AlwaysMultiURIParser)

Plan for OpenAPI integration:
Once this is merged, the OAS3 branch should basically contain
1. The OpenAPIOperation class
2. The OpenAPIURIParser class
3. requestBody form validation
4. modifications to apis/abstract to get the spec version, and instantiate the correct operation class

### CODE REVIEW

**Comment by dtkav** [comment]
Landing #641 should make this diff a little smaller.
I'll remove the WIP from this PR once that's in.

**Comment by jmcs** [comment]
@dtkav can you resolve the conflicts?

File: tests/api/test_bootstrap.py [hunk 1]
```diff
@@ -41,10 +41,10 @@ def test_app_with_different_server_option(simple_api_spec_dir, spec):
 
 
 def test_app_with_different_uri_parser(simple_api_spec_dir):
-    from connexion.decorators.uri_parsing import Swagger2URIParser
```
Comment (dtkav) [inline tests/api/test_bootstrap.py]: I'll factor this out into a separate PR

**Comment by dtkav** [comment]
#653 will make this smaller (factors out switch to openapi_spec_validator)

File: connexion/operations/swagger2.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
+import logging
```
Comment (JuxhinDB) [inline connexion/operations/swagger2.py]: Curious, was this just stripped out from an older file into its own file to ensure backwards compatibility?

**Comment by jmcs** [comment]
:+1:

File: connexion/operations/swagger2.py (hunk 2 above)
Comment (dtkav) [reply connexion/operations/swagger2.py]: Yup, the swagger2.py file is for backwards compatibility with the swagger2 spec. The idea with this PR is to create an Abstract Base Class (abc), which is essentially the interface that describe what an "Operation" is. Previously we only had one implementation of this interface (swagger 2 Operation), so it wasn't well defined. Now, after this refactor, the OpenAPIOperation can implement the exact same interface as a Swagger2Operation. Together they encapsulate the differences between the Swagger2 spec and the OpenAPI3 spec. Hope that makes sense.

**Pull request merged**